src/test/java/com/techsophy/tsf/services/gateway/config/AwgmentOAuthRequestResolverTest.java

```diff
@@ -0,0 +1,225 @@
+package com.techsophy.tsf.services.gateway.config;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.mock.http.server.reactive.MockServerHttpRequest;
+import org.springframework.mock.web.server.MockServerWebExchange;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ReactiveClientRegistrationRepository;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.endpoint.OAuth2AuthorizationRequest;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.util.LinkedMultiValueMap;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+import java.util.List;
+import java.util.Map;
+import static com.techsophy.tsf.services.gateway.constants.GatewayConstants.REALMS;
+import static com.techsophy.tsf.services.gateway.constants.GatewayConstants.X_TENANT;
+import static com.techsophy.tsf.services.gateway.constants.GatewayTestConstants.*;
+import static org.mockito.ArgumentMatchers.anyString;
+
+@ExtendWith(MockitoExtension.class)
+class AwgmentOAuthRequestResolverTest
+{
+    @Mock
+    ReactiveClientRegistrationRepository repo;
+    @InjectMocks
+    AwgmentOAuthRequestResolver awgmentOAuthRequestResolver;
+
+    @BeforeEach
+    void beforeAll()
+    {
+        ReflectionTestUtils.setField(awgmentOAuthRequestResolver,"defaultRealmName","techsophy-platform");
+        ReflectionTestUtils.setField(awgmentOAuthRequestResolver,"keycloakIssuerUri"," https://keycloak-tsplatform.techsophy.com/auth/realms/");
+    }
+
+    @Test
+    void resolveWebExchangeWithHeaderInQueryParamTest()
+    {
+        LinkedMultiValueMap<String, String> headers = new LinkedMultiValueMap<>(Map.of());
```

Comment:
Please show it to me that it is working as intended as I am not able to follow how this would work without mock client repo